src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -620,11 +621,21 @@ db::DataBox<tmpl::list<Tags...>>::mutate_mutable_subitems(
  *
  * \example
  * \snippet Test_DataBox.cpp databox_mutate_example
+ *
+ * The `invokable` may have function return values, and any returns are
+ * forwarded as returns to the `db::mutate` call.
+ *
+ * \warning Using `db::mutate` returns to obtain non-const references or
+ * pointers to box items is potentially very dangerous. The \ref DataBoxGroup
+ * "DataBox" cannot track any subsequent changes to quantities that have been
+ * "unsafely" extracted in this manner, so this should be done only when no
```

Comment:
Should we word this more strongly as "we consider it undefined behavior to return any data from the DataBox from the function", since that's basically the realm that the programmer is in at that point :P It also means that we have some freedom to change DataBox implementations going forward. This would mean we still shouldn't return the lock pointers this way, mostly to keep things in the "defined behavior" realm. I think evening returning reference to const data members can be bad, but I'm not sure it's any worse than using a `db::get` on a dependent compute tag before a `db::mutate` call and having

Reply:
Should we (can we?) define a limited regime of "defined behavior" to extract things like a lock pointer, which we guarantee to retain in future DataBox changes and perhaps add a test for? I think this is what the current comment essentially says: "...[All is fine as long as] no compute or reference tags will access the unsafely extracted tag."

Reply:
I think the current guarantee is that such an extraction/pointer is safe as long as:
- the tag is a simple tag
- no compute or reference tags depend on the tag
- the DataBox is not manipulated in such a way that the memory address of the simple tag's data is changed (no copy, create_from, or post-serialization destruction is done)
- the owning DataBox is not migrated/serialized away
- the extracted data is not serialized